docs_user/modules/openstack-planning.adoc

```diff
@@ -201,10 +201,132 @@ placement of the OpenStack services.
+pods that are part of the service. These IP addresses are managed by the
+configured OpenShift networking provider, e.g. `OVN-Kubernetes`. OpenStack
+services may also define `Service` endpoints for other services to use (with
+attached DNS names or without any).
+
+Since some control plane services need to talk to other services that are not
+part of the OpenShift cluster, they need a way to connect to additional
+(isolated) networks that host EDPM nodes.
+
+This is done using `NetworkAttachmentDefinition` resources. These
+`NetworkAttachmentDefinition` CRs refer to CNI plugins that are installed in
+the OpenShift nodes, and the CNIs are responsible for creating the network
+interfaces in the nodes.
+
+Once the `NetworkAttachmentDefinition` CRs are created, the OpenStack services
+can refer to them in their pod specifications, and the OpenShift platform will
+take care of creating the network interfaces in the nodes and attaching them to
+the pods.
+
+CNI plugins are responsible for implementing the *pod* network interfaces. To
+be able to communicate with other services in the network, the newly created
+interface should be plugged into a bridge that is connected to the physical
+network.
+
+This bridge is not managed by the CNI plugin. It is the responsibility of
+another component running in the OpenShift node, specifically, `nmstate`.
+
+`nmstate` is a daemon that runs in the OpenShift nodes and is responsible for
+the OpenShift *host* networking configuration. It will create necessary
+bridges, VLAN interfaces, etc. which are then used by the CNI plugins to
+interconnect pods with the physical network.
+
+The `nmstate` daemon is deployed as a `DaemonSet` in the OpenShift cluster, and
+receives its configuration from the `NodeNetworkConfigurationPolicy` CRs.
+
+It is important to align the `NodeNetworkConfigurationPolicy` CRs with isolated
+networks used by the OpenStack services, e.g. to align the VLANs used by the
+`NetworkAttachmentDefinition` CRs with the VLANs used by the
+`NodeNetworkConfigurationPolicy` CRs.
+
+`NetworkAttachmentDefinition` CRs also control IPAM allocations for control
+plane services. It is of utmost importance to reconcile
+`NetworkAttachmentDefinition` specifications with the network infrastructure in
+the data center used by the adopted OpenStack deployment.
+
+New podified control plane deployment may require new IPAM allocations, and
+perhaps even more addresses than the original deployment. Depending on the
+existing subnet configuration, particular strategies may have to be followed,
+as explained below.
+
+// TODO: explain how `IPAddressPool` CRs are used to allocate IP addresses for
+// load balancers.
+
+// TODO: explain how networking is configured through NodeSet CRs,
+// edpm_network_config.
+
+// TODO: explain how `NetConfig` CRs are used to configure the networking for
+// EDPM nodes.
+
+// TODO: Anything else to mention here?..
```

Comment:
should You maybe also mention about MetalLB and it's L2 and L3 modes (which may be needed for e.g. BGP deployments)?

Reply:
I'll add a todo to consider it for now; but I am not sure if this concern is specific to adoption (I'll ripping off most details that are not adoption specific in the later versions.)